source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -416,6 +416,16 @@ void StreamRateLimiter::onTokenTimer() {
   ENVOY_LOG(trace, "limiter: timer wakeup: buffered={}", buffer_.length());
   Buffer::OwnedImpl data_to_write;
 
+  if (!saw_data_) {
```

Comment:
I think it would be even more clear if the comment linked the fact that the token_timer_ will not fire, and thus this callback will not be called, until after the first time writeData is called, and thus the timer fired. Or something to the effect?